### PR TITLE
Feature/tg 233/archive buddies

### PR DESCRIPTION
**Pull request by victor-paumier**

### CODE REVIEW

File: src/Command/FixBDDCommand.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        /** @var EntityManager $em */
+        $em = $this->getContainer()->get('doctrine.orm.entity_manager');
+
+        $userRepo = $em->getRepository(User::class);
+        $buddyRepo = $em->getRepository(Buddy::class);
+
+
```
Comment (JeremieSamson) [inline src/Command/FixBDDCommand.php]: remove line plz

File: src/Command/FixBDDCommand.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        $userRepo->createQueryBuilder('u')
+            ->update()
+            ->set('u.isAskNextSemester', 1)
+            ->where('u.archive = :archive')
+            ->andWhere('u.isLocal = :isLocal')
+            ->setParameter('archive', true)
+            ->setParameter('isLocal', true)
+            ->getQuery()->execute();
+
```
Comment (JeremieSamson) [inline src/Command/FixBDDCommand.php]: remove line plz

File: src/Command/FixBDDCommand.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            ->distinct()
+            ->innerJoin('buddy.mentee', 'mentee')
+            ->innerJoin('buddy.mentor', 'mentor')
+            ->where('mentee.archive = :archived')
+            ->orWhere('mentor.archive = :archived')
+            ->setParameter('archived', true)
+            ->getQuery()->getArrayResult();
+
+        $qb = $buddyRepo->createQueryBuilder('b');
+        $qb->update()
```
Comment (JeremieSamson) [inline src/Command/FixBDDCommand.php]: Why not on same line ?

File: src/Controller/BuddyController.php [hunk 4]
```diff
@@ -38,8 +38,12 @@ public function index(SectionRepository $sectionRepo): Response
...
      * @Route("/list_buddies", name="buddy_list", options={ "expose" : true }, condition="request.isXmlHttpRequest()")
      */
-    public function getListBuddies(Request $request, UserInterface $user, BuddyRepository $buddyRepo, TranslatorInterface $translator): JsonResponse
-    {
+    public function getListBuddies(
+        Request $request,
+        UserInterface $user,
+        BuddyRepository $buddyRepo,
+        TranslatorInterface $translator
+    ): JsonResponse {
```
Comment (JeremieSamson) [inline src/Controller/BuddyController.php]: :medal_military:

File: src/Command/FixBDDCommand.php (hunk 3 above)
Comment (victor-paumier) [reply src/Command/FixBDDCommand.php]: need qb for $qb->expr()->in

File: src/Command/FixBDDCommand.php (hunk 1 above)
Comment (victor-paumier) [reply src/Command/FixBDDCommand.php]: done

File: src/Command/FixBDDCommand.php (hunk 2 above)
Comment (victor-paumier) [reply src/Command/FixBDDCommand.php]: done

**Pull request merged**